### PR TITLE
Work on react-ui workplan functionalities

### PR DESCRIPTION
**Pull request by Pkom17**

### CODE REVIEW

**Comment by mozzy11** [comment]
and also in general , we put all the REST controllers under controller/rest/*** to easily identify the REST controllers from the JSP controllers

File: src/main/java/org/openelisglobal/workplan/controller/WorkplanByTestRestController.java [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+
+	private static boolean HAS_NFS_PANEL = false;
+
+	static {
+		HAS_NFS_PANEL = ConfigurationProperties.getInstance().isPropertyValueEqual(Property.CONDENSE_NFS_PANEL, "true");
+	}
+
+	@GetMapping(value = "test-list", produces = MediaType.APPLICATION_JSON_VALUE)
+	@ResponseBody
+	private List<IdValuePair> getTestDropdownList(HttpServletRequest request) {
```
Comment (mozzy11) [inline src/main/java/org/openelisglobal/workplan/controller/WorkplanByTestRestController.java:82]: can you put thiese common Lists under [common](https://github.com/I-TECH-UW/OpenELIS-Global-2/blob/2.8/src/main/java/org/openelisglobal/common/rest/DisplayListController.java)

File: src/main/java/org/openelisglobal/workplan/controller/WorkplanByTestSectionRestController.java [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+	@Autowired
+	private SampleQaEventService sampleQaEventService;
+	
+	@Autowired
+	private RoleService roleService;
+	@Autowired
+	private UserService userService;
+
+
+	@GetMapping(value = "test-sections", produces = MediaType.APPLICATION_JSON_VALUE)
```
Comment (mozzy11) [inline src/main/java/org/openelisglobal/workplan/controller/WorkplanByTestSectionRestController.java:74]: can you put thiese common Lists under [common](https://github.com/I-TECH-UW/OpenELIS-Global-2/blob/2.8/src/main/java/org/openelisglobal/common/rest/DisplayListController.java)

File: src/main/java/org/openelisglobal/workplan/controller/PrintWorkplanReportController.java [hunk 3]
```diff
@@ -107,6 +108,70 @@ public void showPrintWorkplanReport(HttpServletRequest request, HttpServletRespo
         }
     }
 
+    @RequestMapping(value = "/rest/printWorkplanReport", method = RequestMethod.POST)
```
Comment (mozzy11) [inline src/main/java/org/openelisglobal/workplan/controller/PrintWorkplanReportController.java:111]: i guess you would want to create a separate REST contoller

File: src/main/java/org/openelisglobal/workplan/controller/WorkplanByPriorityRestController.java [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+public class WorkplanByPriorityRestController extends WorkplanRestController {
+
+    @Autowired
+    private AnalysisService analysisService;
+    @Autowired
+    private SampleQaEventService sampleQaEventService;
+    @Autowired
+    private UserService userService;	
+
+	@GetMapping(value = "priorities", produces = MediaType.APPLICATION_JSON_VALUE)
```
[the remark this answers is not in the document]
Comment (Pkom17) [reply src/main/java/org/openelisglobal/workplan/controller/WorkplanByPriorityRestController.java]: solved

File: src/main/java/org/openelisglobal/workplan/controller/WorkplanByPanelRestController.java [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+	@Autowired
+	private AnalysisService analysisService;
+	@Autowired
+	private PanelItemService panelItemService;
+	@Autowired
+	private SampleQaEventService sampleQaEventService;
+	@Autowired
+	private UserService userService;
+
+	@GetMapping(value = "panels", produces = MediaType.APPLICATION_JSON_VALUE)
```
[the remark this answers is not in the document]
Comment (Pkom17) [reply src/main/java/org/openelisglobal/workplan/controller/WorkplanByPanelRestController.java]: solved

File: src/main/java/org/openelisglobal/workplan/controller/WorkplanByTestRestController.java (hunk 1 above)
Comment (Pkom17) [reply src/main/java/org/openelisglobal/workplan/controller/WorkplanByTestRestController.java]: solved

File: src/main/java/org/openelisglobal/workplan/controller/PrintWorkplanReportController.java (hunk 3 above)
Comment (Pkom17) [reply src/main/java/org/openelisglobal/workplan/controller/PrintWorkplanReportController.java]: done

**Comment by mozzy11** [comment]
@Pkom17 can you fix these merge conflicts

**Pull request merged**